api/lql.go

```diff
@@ -18,24 +18,177 @@
 
 package api
 
+import (
+	"encoding/json"
+	"net/url"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	reLQL                  string = `(?ms)^(\w+)\([^)]+\)\s*{`
+	LQLQueryTranslateError string = "unable to translate query blob"
+)
+
+type LQLQuery struct {
+	ID             string `json:"LQL_ID,omitempty"`
+	StartTimeRange string `json:"START_TIME_RANGE,omitempty"`
+	EndTimeRange   string `json:"END_TIME_RANGE,omitempty"`
+	QueryText      string `json:"QUERY_TEXT"`
+	// QueryBlob is a special string that supports type conversion
+	// back and forth from LQL to JSON
+	QueryBlob string `json:"-"`
+}
+
+func (q *LQLQuery) Translate() (returnError error) {
+	// if QueryText is populated; return
+	if q.QueryText != "" {
+		return
+	}
+
+	// if QueryBlob is JSON
+	var t LQLQuery
+
+	if err := json.Unmarshal([]byte(q.QueryBlob), &t); err == nil {
+		if q.StartTimeRange == "" {
+			q.StartTimeRange = t.StartTimeRange
+		}
+		if q.EndTimeRange == "" {
+			q.EndTimeRange = t.EndTimeRange
+		}
+		q.QueryText = t.QueryText
+		return
+	}
+
+	// if QueryBlob is LQL
+	if matched, _ := regexp.MatchString(reLQL, q.QueryBlob); matched {
+		q.QueryText = q.QueryBlob
+		return
+	}
+
+	return errors.New(LQLQueryTranslateError)
+}
+
+func (q *LQLQuery) Validate(allowEmptyTimes bool) error {
```

Comment:
Nit!! 

For readability in Go we prefer to create multiple functions to be explicit about their use,
for example, this `Validate()` function could have been two functions:

- `Validate()` or `ValidateEnforceTimes()`
- `ValidateAllowEmptyTimes()`

This will avoid us devs having to go look at the function and understand what
the boolean argument does.

Reply:
This passes through to multiple functions, so instead of 2 functions you'll have 4 functions that do basically the same thing...